src/main/java/picard/analysis/CollectUmiPrevalenceMetrics.java

```diff
@@ -0,0 +1,181 @@
+import org.broadinstitute.barclay.argparser.ExperimentalFeature;
+import org.broadinstitute.barclay.help.DocumentedFeature;
+import picard.cmdline.CommandLineProgram;
+import picard.cmdline.StandardOptionDefinitions;
+import picard.cmdline.programgroups.DiagnosticsAndQCProgramGroup;
+import picard.filter.CountingFilterWrapper;
+import picard.filter.CountingMapQFilter;
+import picard.filter.CountingPairedFilter;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.*;
+import java.util.stream.IntStream;
+
+import static picard.cmdline.StandardOptionDefinitions.MINIMUM_MAPPING_QUALITY_SHORT_NAME;
+
+/**
+ * A CLP that, tallies the number of UMIs in a duplicate set across the entire bam and produces a histogram.
+ *
+ * @author Yossi Farjoun
+ */
+@DocumentedFeature
+@ExperimentalFeature
+@CommandLineProgramProperties(
+        summary = CollectUmiPrevalenceMetrics.USAGE_SUMMARY + CollectUmiPrevalenceMetrics.USAGE_DETAILS,
+        oneLineSummary = CollectUmiPrevalenceMetrics.USAGE_SUMMARY,
+        programGroup = DiagnosticsAndQCProgramGroup.class
+)
+public class CollectUmiPrevalenceMetrics extends CommandLineProgram {
+
+    static final String USAGE_SUMMARY = "Tally the counts of UMIs in duplicate sets within a bam. \n";
+    static final String USAGE_DETAILS = "<p>" +
+            "This tool collects the Histogram of the number of duplicate sets that contain a given number of UMIs. " +
+            "Understanding this distribution can help understand the role that the UMIs have in the determination of " +
+            "consensus sets, the risk of UMI collisions, and of spurious reads that result from uncorrected UMIs.";
+
+
+    @Argument(shortName = StandardOptionDefinitions.INPUT_SHORT_NAME, doc = "Input (indexed) BAM/CRAM file.")
+    public File INPUT;
+
+    @Argument(shortName = StandardOptionDefinitions.OUTPUT_SHORT_NAME, doc = "Write metrics to this file")
+    public File OUTPUT;
+
+    @Argument(shortName = MINIMUM_MAPPING_QUALITY_SHORT_NAME, doc = "minimal value for the mapping quality of the reads to be used in the estimation.", optional = true)
+    public Integer MINIMUM_MQ = 30;
+
+    @Argument(doc = "Barcode SAM tag.", optional = true)
+    public String BARCODE_TAG = "RX";
+
+    @Argument(doc = "Barcode Quality SAM tag.", optional = true)
+    public String BARCODE_BQ = "QX";
+
+    @Argument(shortName = "BQ", doc = "minimal value for the base quality of all the bases in a molecular barcode, for it to be used.", optional = true)
+    public Integer MINIMUM_BARCODE_BQ = 30;
+
+    @Argument(shortName = "FUR", doc = "Whether to filter unpaired reads from the input.", optional = true)
+    public boolean FILTER_UNPAIRED_READS = true;
+
+    @Argument(
+            fullName = "PROGRESS_STEP_INTERVAL",
+            doc = "The interval between which progress will be displayed.",
+            optional = true
+    )
+    public int PROGRESS_STEP_INTERVAL = 1_000_000;
```

Comment:
I appreciate the use of _ here to make this readable.  Great language feature...